test/functorch/test_control_flow.py

```diff
@@ -1217,75 +1254,63 @@ def forward(self):
             """\
 def forward(self, arg0_1, arg1_1, arg2_1, arg3_1):
     mul = torch.ops.aten.mul.Tensor(arg2_1, arg1_1);  arg2_1 = mul = None
-    mul_1 = torch.ops.aten.mul.Tensor(arg0_1, arg1_1);  arg0_1 = arg1_1 = None
-    zeros_like = torch.ops.aten.zeros_like.default(arg3_1, pin_memory = False);  arg3_1 = None
-    return [None, mul_1, zeros_like]""",
+    mul_1 = torch.ops.aten.mul.Tensor(arg0_1, arg1_1);  arg0_1 = None
+    zeros_like = torch.ops.aten.zeros_like.default(arg1_1, pin_memory = False);  arg1_1 = None
+    zeros_like_1 = torch.ops.aten.zeros_like.default(arg3_1, pin_memory = False);  arg3_1 = None
+    return [zeros_like, mul_1, zeros_like_1]""",
         )
 
         self.assertExpectedInline(
             joint_false_graph.code.strip(),
             """\
 def forward(self, arg0_1, arg1_1, arg2_1, arg3_1):
     mul = torch.ops.aten.mul.Tensor(arg3_1, arg1_1);  arg3_1 = mul = None
-    mul_1 = torch.ops.aten.mul.Tensor(arg0_1, arg1_1);  arg0_1 = arg1_1 = None
-    zeros_like = torch.ops.aten.zeros_like.default(arg2_1, pin_memory = False);  arg2_1 = None
-    return [None, zeros_like, mul_1]""",
+    mul_1 = torch.ops.aten.mul.Tensor(arg0_1, arg1_1);  arg0_1 = None
+    zeros_like = torch.ops.aten.zeros_like.default(arg1_1, pin_memory = False);  arg1_1 = None
+    zeros_like_1 = torch.ops.aten.zeros_like.default(arg2_1, pin_memory = False);  arg2_1 = None
+    return [zeros_like, zeros_like_1, mul_1]""",
         )
 
     @unittest.skipIf(not torch.cuda.is_available(), "Test requires CUDA.")
-    def test_cond_autograd_zeros_unused_branch_complex(self):
+    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    # @parametrize("compile_mode", ["none", "eager", "compile", "compile_dynamic_shape"])
```

Comment:
Yes, the device is optional and does not really matter. I removed it.
In case of `compile` or `compile_dynamic_shape`, the error is
`torch._inductor.exc.LoweringException: NotImplementedError: get_size() is not implemented by <class 'torch._inductor.ir.NoneAsConstantBuffer'>!`

I added a TODO with a comment on this.

Reply:
This seems to be caused by a None output is returned from somewhere ( ir.Conditional I guess). Looking at the model, it doesn't seem that we're returning a None output? Maybe paste the full log? This worth investigating more.